plugins/wazuh-core/public/services/server-security/ui/components/element.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Wazuh app - Button with Wazuh API permissions and/or roles required to be useful
+ * Copyright (C) 2015-2022 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React, { Fragment } from 'react';
+import { EuiToolTip, EuiSpacer } from '@elastic/eui';
+import { ServerPermissionsFormatted } from './format';
+export interface IUserPermissionsObject {
+  action: string;
+  resource: string;
+}
+export type TUserPermissionsFunction = (props: any) => TUserPermissions;
+export type TUserPermissions = (string | IUserPermissionsObject)[] | null;
+export type TUserRoles = string[] | null;
+export type TUserIsAdministrator = string | null;
+export type TUserRolesFunction = (props: any) => TUserRoles;
+
+export interface ServerElementPermissionsProps {
+  permissions?: TUserPermissions | TUserPermissionsFunction;
+  administrator?: boolean;
+  tooltip?: any;
+  children: React.ReactElement;
+  getAdditionalProps?: (disabled: boolean) => {
+    [prop: string]: any;
+  };
+}
+
+export const ServerElementPermissions = ({
+  children,
+  permissions = null,
```

Comment:
```suggestion
  permissions: Function | object | null = null,
```

Reply:
The type for `permissions` are defined here https://github.com/wazuh/wazuh-dashboard-plugins/pull/7026/files#diff-db1ea29a2b610976d178ef552064997488c67d5efde4841c737f533039dbdd6eR45